apps/api/src/queries/buildViewModelForResource/viewModels/connection-record-for-resource.view-model.ts

```diff
@@ -0,0 +1,70 @@
+import {
+    EdgeConnectionContextType,
+    EdgeConnectionMemberRole,
+    IEdgeConnectionContext,
+    ResourceCompositeIdentifier,
+    ResourceType,
+} from '@coscrad/api-interfaces';
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import { buildMultilingualTextWithSingleItem } from '../../../domain/common/build-multilingual-text-with-single-item';
+import { MultilingualText } from '../../../domain/common/entities/multilingual-text';
+import buildDummyUuid from '../../../domain/models/__tests__/utilities/buildDummyUuid';
+import { CoscradDataExample } from '../../../test-data/utilities';
+import { DTO } from '../../../types/DTO';
+
+@CoscradDataExample<ConnectionRecordForResourceViewModel>({
+    example: {
+        id: buildDummyUuid(1),
+        note: buildMultilingualTextWithSingleItem('test note'),
+        selfContext: {
+            type: EdgeConnectionContextType.general,
+        },
+        otherCompositeIdentifier: {
+            type: ResourceType.audioItem,
+            id: buildDummyUuid(123),
+        },
+        otherContext: {
+            type: EdgeConnectionContextType.general,
+        },
+        role: EdgeConnectionMemberRole.from,
+    },
+})
+export class ConnectionRecordForResourceViewModel {
```

Comment:
I'm wondering if the update method should actually be on the note query repository. But this is something we need to introduce. The only thing we really need is a way to get the collection name `widget__VIEWS` from the resource type `widget`. This can be done (as currently) using convention or by a lookup \ dynamic registration.